src/datepicker/datepicker-tools.ts

```diff
@@ -78,11 +78,14 @@ export function generateSelectBoxYears(date: NgbDate, minDate: NgbDate, maxDate:
 }
 
 export function nextMonthDisabled(calendar: NgbCalendar, date: NgbDate, maxDate: NgbDate) {
-  return maxDate && calendar.getNext(date, 'm').after(maxDate);
+  const nextDate = calendar.getNext(date, 'm');
```

Comment:
FYI, can be a bit shorter, but I have no preference really. Change if you like it
```ts
  const nextDate = Object.assign(calendar.getNext(date, 'm'), {day: 1});
```